tests/integration/distribution.go

```diff
@@ -1060,6 +1060,114 @@ func (s *CCVTestSuite) TestAllocateTokensToValidator() {
 	}
 }
 
+// TestAllocateTokensToConsumerValidatorsWithDifferentValidatorHeights tests `AllocateTokensToConsumerValidators` with
+// consumer validators that have different heights. Specifically, test that validators that have been consumer validators
+// for some time receive rewards, while validators that recently became consumer validators do not receive rewards.
+func (s *CCVTestSuite) TestAllocateTokensToConsumerValidatorsWithDifferentValidatorHeights() {
+	// Note this test is an adaptation of a `TestAllocateTokensToConsumerValidators` testcase.
+	providerKeeper := s.providerApp.GetProviderKeeper()
+	distributionKeeper := s.providerApp.GetTestDistributionKeeper()
+	bankKeeper := s.providerApp.GetTestBankKeeper()
+
+	chainID := s.consumerChain.ChainID
+
+	tokens := sdk.DecCoins{sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, math.LegacyNewDecFromIntWithPrec(math.NewInt(999), 2))}
+	rate := sdk.OneDec()
+	expAllocated := sdk.DecCoins{sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, math.LegacyNewDecFromIntWithPrec(math.NewInt(999), 2))}
+
+	ctx, _ := s.providerCtx().CacheContext()
+	// If the provider chain has not yet reached `GetNumberOfEpochsToStartReceivingRewards * GetBlocksPerEpoch` block height,
+	// then all validators receive rewards. In this test, we want to check whether validators receive rewards or not based on how
```

Comment:
This comment is slightly confusing. What does it imply if the provider chain hasn't reached `GetNumberOfEpochsToStartReceivingRewards * GetBlocksPerEpoch`? Why should the validators get rewards based on that?

Reply:
I have a comment in this PR's description that I slightly updated about this:
> Note that at the beginning of a provider chain (e.g., when the block height of the provider is still small), no validator could have been a consumer validator for more than `NumberOfEpochsToStartReceivingRewards` epochs and hence in this case we distribute rewards to validators, even though they might have not been validating the consumer chain for more than a single epoch.

I introduced the `IsEligibleForConsumerRewards` as you suggested and added that comment there as well to make it clear. If this did not clarify things, let me know and I'll update this.